crates/ruma-events/src/event_kinds.rs

```diff
@@ -9,9 +9,16 @@ use crate::{
     StateEventContent, Unsigned,
 };
 
-/// A basic event – one that consists only of it's type and the `content` object.
+/// A global account data event.
 #[derive(Clone, Debug, Event)]
-pub struct BasicEvent<C: BasicEventContent> {
+pub struct GlobalAccountDataEvent<C: BasicEventContent> {
```

Comment:
I wonder whether we should just go all the way and create separate traits too.

Reply:
Would make the whole interface feel a bit more consistent.

Reply:
There's some merit (at least theoretically) to not being able to construct `GlobalAccountDataEvent<DummyEventContent>`. And the amount of extra code should be fairly small.

Reply:
Ah right!

Reply:
Actually this trait is used for other events. But we could use implement both this trait and new ones for the account data events (maybe that's what you meant).